node/sync/src/service.rs

```diff
@@ -576,6 +576,21 @@ impl SyncService {
 
         // trigger retry after failure
         if let SyncState::Failed { .. } = controller.get_status() {
+            let goal = controller.get_sync_info().goal;
+
+            match maybe_range {
+                Some((start, end)) => {
+                    if goal.is_all_chunks() || start != goal.index_start || end != goal.index_end {
+                        bail!("Cannot change failed file sync to chunks sync");
+                    }
+                }
+                None => {
+                    if !goal.is_all_chunks() {
+                        bail!("Cannot change failed chunks sync to file sync");
+                    }
+                }
+            }
+
             controller.reset();
         }
 
```

Comment:
Update controller's range if the old status is completed, the old goal is not all chunks and the new goal is different from the old.